GEON-PPANG-iOS/Presentation/BakeryList/ViewControllers/SortBottomSheetViewController.swift

```diff
@@ -0,0 +1,226 @@
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(dismissingComponentTapped(_:)))
+        dimmedView.addGestureRecognizer(tapGesture)
+        dimmedView.isUserInteractionEnabled = true
+    }
+    
+    private func showBottomSheet() {
+        UIView.animate(withDuration: 0.2) {
+            self.dimmedView.do {
+                $0.backgroundColor = .black.withAlphaComponent(0.4)
+            }
+            self.bottomSheetView.snp.updateConstraints {
+                $0.height.equalTo(self.bottomSheetHeight)
+            }
+            self.view.layoutIfNeeded()
+        }
+    }
+    
+    // MARK: - Action Helper
+    
+    @objc
+    private func dismissingComponentTapped(_ tapRecognizer: UITapGestureRecognizer) {
+        dismissBottomSheetViewController()
+    }
+    
+    // MARK: - Custom Method
+    
+    private func dismissBottomSheetViewController() {
+        UIView.animate(withDuration: 0.2) {
+            self.dimmedView.do {
+                $0.backgroundColor = .black.withAlphaComponent(0.0)
+            }
+            self.bottomSheetView.snp.makeConstraints {
+                $0.height.equalTo(0)
+            }
+            self.view.layoutIfNeeded()
+        } completion: { _ in
+            self.dismiss(animated: false)
+        }
+    }
+    
+    // TODO: Utils 로 빠진 함수 사용
+    func updateCollectionViewConstraint(of collectionView: UICollectionView) {
+        let height = collectionView.collectionViewLayout.collectionViewContentSize.height
+        guard height != 0 else { return }
+        collectionView.snp.updateConstraints {
+            $0.height.equalTo(height)
+        }
+    }
+    
+}
+
+// MARK: - UICollectionViewDelegate extension
+
+extension SortBottomSheetViewController: UICollectionViewDelegate {
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        switch indexPath.item {
+        case 0: selectedSortBy = .byDefault
+        case 1: selectedSortBy = .byReviews
+        default: dismissBottomSheetViewController()
+        }
+        dataBind?(selectedSortBy)
+//        print("selectedSortBy - sheet", selectedSortBy)
+        dismissBottomSheetViewController()
+    }
+    
+}
+
+// MARK: - UICollectionViewDataSource extension
+
+extension SortBottomSheetViewController: UICollectionViewDataSource {
+    
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return SortBakery.allCases.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell: SortBakeryCollectionViewCell = collectionView.dequeueReusableCell(for: indexPath)
+        let data = SortBakery.allCases.map { $0.description }
+        cell.configureLabel(to: data[indexPath.item])
+//        switch indexPath.item {
+//        case 0:
+//            if selectedSortBy == .byDefault
+//        }
```

Comment:
p2: what is it ?

Reply:
[9a003e6](https://github.com/GEON-PPANG/GEON-PPANG-iOS/pull/57/commits/9a003e664834cbfac2b64abc730b5f3409e23684)
ㅋㅋ ㅎ